### PR TITLE
chore(docs-app): add table of contents to individual pages 

### PR DESCRIPTION
**Pull request by Narretz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
chore/feature for the docs app

**What is the current behavior? (You can also link to an open issue here)**
Individual docs pages don't have a toc, which makes it difficult to navigate longer pages

**What is the new behavior (if this is a feature change)?**
They get one. This also makes it easier to find the hash links for individual sections.

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

Initially discussed here: https://github.com/angular/dgeni-packages/issues/81

This PR adds a table of contents to every individual page of the docs, i.e. API, guide - everything.
It's based on the `id` of the `<h*>` elements, and expects a hierarchical usage of the headings. There shiuld only be one h1 on the page, which is the page title. h2 to h5 should be used in hierarchy, i.e. h3 only as child of h2 and so on. Several docs pages and templates have been adjusted to follow these rules. Also, a few manual tocs have been removed.

In general, the tocs are not very long, so they are always displayed in full (exception is the migration guide), but a button has been added to hide the toc completely. No option to hide single branches exist, but most branches are rather short anyway. Toc with only a single h2 or fewer are not displayed.

Issues: 

- Headings in tags that are collected for the module overview pages (e.h. @knownIssue) create strange tocs
- Examples have inconsistent headings
- Sections / Headings with the same name generate incorrect links in the toc (e.g ngAria guide) -> should be handled by dgeni

short
fireshot screen capture 038 - angularjs_ api_ angular_copy

medium
fireshot screen capture 035 - angularjs_ api_ select

long
fireshot screen capture 037 - angularjs_ api_ compile

### CODE REVIEW

File: src/ngCookies/cookies.js [hunk 1]
```diff
@@ -4,12 +4,9 @@
  * @ngdoc module
  * @name ngCookies
  * @description
- *
- * # ngCookies
- *
+ * *
```
Comment (petebacondarwin) [inline src/ngCookies/cookies.js]: Was there supposed to be a newline in here somewhere?

File: src/ngMock/angular-mocks.js [hunk 2]
```diff
@@ -2356,15 +2356,10 @@ angular.mock.$ComponentControllerProvider = ['$compileProvider',
...
  * @description
  *
- * # ngMock
- *
  * The `ngMock` module provides support to inject and mock AngularJS services into unit tests.
  * In addition, ngMock also extends various core AngularJS services such that they can be
  * inspected and controlled in a synchronous manner within test code.
  *
- *
- * <div doc-module-components="ngMock"></div>
```
Comment (petebacondarwin) [inline src/ngMock/angular-mocks.js]: Are these no longer needed?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
LGTM except for a complete lack of tests :-)

File: src/ngMock/angular-mocks.js (hunk 2 above)
Comment (Narretz) [reply src/ngMock/angular-mocks.js]: No, I think they are left over from a time when these were generated differently. The directive doesn't exist anymore. dgeni renders the components during the docs build.

**Review by petebacondarwin — APPROVED** [review]
LGTM (except for a naughty little `fdescribe` in there)!

**Comment by Narretz** [comment]
Finally merged. I had to remove the unit test because the matching of the DOM strings didn't work because browsers order the attributes differently. :<

**Comment by Narretz** [comment]
Finally merged. I had to remove the unit test because the matching of the DOM strings didn't work because browsers order the attributes differently. :<

**Pull request merged**